hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -28,6 +28,65 @@ def _filter_missing_method(self, filter_missing: bool, name: str, ret_type: Hail
             collection = self.filter(hl.is_defined)
         return collection._method(name, ret_type, *args)
 
+    def __getattr__(self, item):
+        return CollectionExpression.__getitem__(self, item)
+
+    @typecheck_method(item=oneof(str))
+    def __getitem__(self, item):
+        """Get a field from each struct in this collection.
+
+        Examples
+        --------
+
+        Note, in the examples below ``hl.eval`` is used to convert a hail
+        expression to a python value so that it prints nicely. It is not
+        necssary to use ``hl.eval`` inside ``annotate_rows`` or similar
+        methods.
+
+        >>> x = hl.array([hl.struct(a='foo', b=3), hl.struct(a='bar', b=4)])
+        >>> hl.eval(x.a)
+        ['foo', 'bar']
+
+        >>> a = hl.array([hl.struct(b=[hl.struct(inner=1),
+        ...                            hl.struct(inner=2)]),
+        ...               hl.struct(b=[hl.struct(inner=3)])])
+        >>> hl.eval(a.b)
+        [[Struct(inner=1), Struct(inner=2)], [Struct(inner=3)]]
+        >>> hl.eval(a.b.inner)
+        [[1, 2], [3]]
+        >>> hl.eval(hl.flatten(a.b).inner)
+        [1, 2, 3]
+        >>> hl.eval(hl.flatten(a.b.inner))
+        [1, 2, 3]
+
+        Parameters
+        ----------
+        item : :obj:`str`
+            Field name
+
+        Returns
+        -------
+        :class:`.CollectionExpression`
+            A collection formed by getting the given field for each struct in
+            this collection
+        """
+
+        etype = self.dtype.element_type
+        if isinstance(etype, hl.tstruct):
+            return hl.struct(
+                **{k: self.map(lambda x: x[k]) for k in etype}).__getitem__(item)
```

Comment:
This converts from the row format to the column format, then it performs the get item. I suppose it can be directly written as:

```
return self.map(lambda x: x[item])
```

Conceptually though, I think of myself as operating on the column formatted data. I'll change to `self.map`.